src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -194,7 +195,12 @@ static RuntimeMetrics()
 
         AppDomain.CurrentDomain.FirstChanceException += (source, e) =>
         {
-            exceptionCounter.Add(1);
+            var tags = new TagList
+            {
+                new("type", e.Exception.GetType().Name),
+            };
+
+            exceptionCounter.Add(1, tags);
```

Comment:
The main scenario is to get a rough idea of the total number and see if it is abnormally high.

For the purpose of counting exception types, there is a concern on the number of types of exceptions in customer's code. If there are too many, it may impact the performance. One key design principle of this library is to minimize resource usage (memory consumption and pressure on garbage collection).

Reply:
Would it be okay to make it possible to customize which tags could be recorded?

Reply:
Separate from the above qn.
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/test/Benchmarks/Metrics/MetricsBenchmarks.cs#L98 Use TagList only if you have more than 3 tags to pass. Else use the `Add(1, tag1)` overloads.

Reply:
@cijothomas done in 9a1ff99. Since there was already a options type for runtime metrics I've added an option to record exception types, defaulting to `false`. Had to make a few changes to the `RuntimeMetrics` class, such as making its static constructor an instance constructor to allow creating the counter taking the options into account.

Reply:
@cijothomas @xiang17 should I close this?

Reply:
@dpk83 Whats your take on this? Should exception type be collected by default vs make it opt-in?

Reply:
> @dpk83 Whats your take on this? Should exception type be collected by default vs make it opt-in?

I think opt-in is fine here as this gives users a little more flexibility.